cfme/configure/configuration/__init__.py

```diff
@@ -800,8 +792,7 @@ def update(self):
         sel.force_navigate("cfg_settings_currentserver_server")
         fill(self.basic_information, self.details)
         sel.click(form_buttons.save)
-        # TODO: Maybe make a cascaded delete on lazycache?
-        fire('server_details_changed')
+        store.current_appliance.server_details_changed()
```

Comment:
I'd say go the extra step and get the self.appliance into the object, we'll need it for navigation anyway.